docs/ref/migration-operations.txt

```diff
@@ -102,9 +102,14 @@ subclass).
 
 .. class:: AlterIndexTogether(name, index_together)
 
-Changes the model's set of custom indexes (the
-:attr:`~django.db.models.Options.index_together` option on the ``Meta``
-subclass).
+Changes the model's set of custom indexes (the ``index_together`` option on the
+``Meta`` subclass).
+
+.. warning::
+
+    ``AlterIndexTogether`` is officially supported only for pre-Django 4.2
+    migration files. For backward compatibility reasons, it's still part of the
+    public API, and there's no plan to deprecate or remove it.
```

Comment:
Don't shoot me 😜... 

Is it worth having an _Unapplied only_ check (perhaps just a warning) on migrations if they use ``AlterIndexTogether`` so folks who didn't read this know not to use it? 

That would trigger if I rolled-back (or rebuilt or ...) as per the previous discussion but that seems OK. 🤔

_No_ might be the right answer 🙂

Reply:
I would say "no" :wink: Django 4.2+ will not generate migrations with `AlterIndexTogether` anymore, and who would add it by hand :shrug: